### PR TITLE
Move ConfiguredGraphFactory into janusgraph-server

### LINKED ISSUE #625: janusgraph-core should not depend on gremlin-server
I think any deps on `gremlin-server` should be pulled out into a new module `janusgraph-server`.

### PR DESCRIPTION
**Pull request by farodin91**
Fixes #625

Signed-off-by: Jan Jansen <jan.jansen@gdata.de>

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there an issue associated with this PR? Is it referenced in the commit message?
- [ ] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [ ] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](https://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?
- [ ] If this PR is a documentation-only change, have you added a `[doc only]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by debasishdebs** [comment]
@farodin91 : Made a first pass at some files will continue reviewing through this. Though an intriguing, thing your recent commit is leading to build failures?

File: janusgraph-server/src/main/java/org/janusgraph/core/CfgJanusGraphFactory.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+     * graph reference tracker, if there.
+     *
+     * <p><b>WARNING: This is an irreversible operation that will delete all graph and index data.</b></p>
+     * @param graph JanusGraph graph database. Can be open or closed.
+     * @throws BackendException If an error occurs during deletion
+     */
+    public static void drop(JanusGraph graph) throws BackendException {
+        Preconditions.checkNotNull(graph);
+        Preconditions.checkArgument(graph instanceof StandardJanusGraph,"Invalid graph instance detected: %s",graph.getClass());
+        final StandardJanusGraph g = (StandardJanusGraph) graph;
```
Comment (debasishdebs) [inline janusgraph-server/src/main/java/org/janusgraph/core/CfgJanusGraphFactory.java:117]: Same comment here as in `JanusGraphFactory` as https://github.com/JanusGraph/janusgraph/pull/2111#discussion_r429775308

File: janusgraph-server/src/main/java/org/janusgraph/graphdb/database/management/CfgManagementSystem.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        this.graph = graph;
+        this.managementLogger = (CfgManagementLogger) graph.getManagementLogger();
+    }
+
+    /**
+     * Upon the open managementsystem's commit, this graph will be asynchronously evicted from the cache on all JanusGraph nodes in your
+     * cluster, once there are no open transactions on this graph on each respective JanusGraph node
+     * and assuming each node is correctly configured to use the {@link org.janusgraph.graphdb.management.JanusGraphManager}.
+     */
+    public void evictGraphFromCache() {
```
Comment (debasishdebs) [inline janusgraph-server/src/main/java/org/janusgraph/graphdb/database/management/CfgManagementSystem.java:41]: Is there any specific reason for moving `evictGraphFromCache` from `ManagementSystem`? It essentially does the same thing as in `ManagementSystem` and doesn't add any additional funtionalies.

**Comment by farodin91** [comment]
> But I thought @farodin91 this PR was to integrate gRPC server into JanusGraph server or just moving ConfiguredFactory from janusgraph-core to janusgraph-server ?

I currently working on this and another PR as preparation to add the gRPC server.

**Comment by farodin91** [comment]
I will come back after we have a management solution in place.

**Pull request closed without merging**